<a href="https://colab.research.google.com/github/Rafarbm/Ciencia_Dados_Aprendizagem_Maquina/blob/main/Painel_suporte.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [21]:
from google.colab import files
import pandas as pd

uploaded = files.upload()

arquivo = next(iter(uploaded))

df = pd.read_csv(arquivo)

df.head()

Saving chamados_suporte_ti.csv to chamados_suporte_ti (2).csv


,id_chamado,data_abertura,setor,categoria,prioridade,status,tempo_resolucao_horas,satisfacao,reaberto,analista
0,CH001,2026-08-20,Financeiro,Acesso,Alta,Resolvido,3.5,5.0,Não,Ana
1,CH002,2026-08-20,Acadêmico,Sistema lento,Média,Resolvido,12.0,4.0,Não,Bruno
2,CH003,2026-08-21,Biblioteca,Impressora,Baixa,Resolvido,30.0,3.0,Sim,Carla
3,CH004,2026-08-21,Financeiro,Erro de sistema,Crítica,Resolvido,2.0,5.0,Não,Ana
4,CH005,2026-08-22,RH,Acesso,Média,Em andamento,NaN,NaN,Não,Diego


In [22]:
import pandas as pd

df = pd.read_csv("/content/chamados_suporte_ti.csv")

df.head()

,id_chamado,data_abertura,setor,categoria,prioridade,status,tempo_resolucao_horas,satisfacao,reaberto,analista
0,CH001,2026-08-20,Financeiro,Acesso,Alta,Resolvido,3.5,5.0,Não,Ana
1,CH002,2026-08-20,Acadêmico,Sistema lento,Média,Resolvido,12.0,4.0,Não,Bruno
2,CH003,2026-08-21,Biblioteca,Impressora,Baixa,Resolvido,30.0,3.0,Sim,Carla
3,CH004,2026-08-21,Financeiro,Erro de sistema,Crítica,Resolvido,2.0,5.0,Não,Ana
4,CH005,2026-08-22,RH,Acesso,Média,Em andamento,NaN,NaN,Não,Diego


In [23]:
print("Dimensões:", df.shape)
print("Colunas:", df.columns.tolist())

df.info()

Dimensões: (18, 10)
Colunas: ['id_chamado', 'data_abertura', 'setor', 'categoria', 'prioridade', 'status', 'tempo_resolucao_horas', 'satisfacao', 'reaberto', 'analista']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18 entries, 0 to 17
Data columns (total 10 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id_chamado             18 non-null     object 
 1   data_abertura          18 non-null     object 
 2   setor                  18 non-null     object 
 3   categoria              18 non-null     object 
 4   prioridade             18 non-null     object 
 5   status                 18 non-null     object 
 6   tempo_resolucao_horas  16 non-null     float64
 7   satisfacao             16 non-null     float64
 8   reaberto               18 non-null     object 
 9   analista               18 non-null     object 
dtypes: float64(2), object(8)
memory usage: 1.5+ KB


In [24]:
df.describe()

,tempo_resolucao_horas,satisfacao
count,16.000000,16.000000
mean,23.000000,3.625000
std,29.648496,1.310216
min,2.000000,1.000000
25%,6.750000,2.750000
50%,13.000000,4.000000
75%,22.500000,5.000000
max,120.000000,5.000000


In [25]:
colunas = [
    "id_chamado",
    "categoria",
    "prioridade",
    "status",
    "tempo_resolucao_horas",
    "satisfacao",
    "reaberto"
]

analise = df[colunas].copy()

analise.head()

,id_chamado,categoria,prioridade,status,tempo_resolucao_horas,satisfacao,reaberto
0,CH001,Acesso,Alta,Resolvido,3.5,5.0,Não
1,CH002,Sistema lento,Média,Resolvido,12.0,4.0,Não
2,CH003,Impressora,Baixa,Resolvido,30.0,3.0,Sim
3,CH004,Erro de sistema,Crítica,Resolvido,2.0,5.0,Não
4,CH005,Acesso,Média,Em andamento,NaN,NaN,Não


In [26]:
resolvidos = analise[analise["status"] == "Resolvido"].copy()

resolvidos

,id_chamado,categoria,prioridade,status,tempo_resolucao_horas,satisfacao,reaberto
0,CH001,Acesso,Alta,Resolvido,3.5,5.0,Não
1,CH002,Sistema lento,Média,Resolvido,12.0,4.0,Não
2,CH003,Impressora,Baixa,Resolvido,30.0,3.0,Sim
3,CH004,Erro de sistema,Crítica,Resolvido,2.0,5.0,Não
5,CH006,Rede,Alta,Resolvido,9.5,2.0,Sim
6,CH007,Hardware,Média,Resolvido,20.0,4.0,Não
7,CH008,Sistema lento,Baixa,Resolvido,45.0,4.0,Não
8,CH009,Erro de sistema,Alta,Resolvido,7.0,5.0,Não
9,CH010,Rede,Crítica,Resolvido,6.0,2.0,Sim
11,CH012,Acesso,Baixa,Resolvido,10.0,5.0,Não


In [27]:
limites = {
    "Crítica": 4,
    "Alta": 8,
    "Média": 24,
    "Baixa": 48
}

resolvidos["limite_sla_horas"] = resolvidos["prioridade"].map(limites)

In [28]:
resolvidos["dentro_sla"] = (
    resolvidos["tempo_resolucao_horas"]
    <= resolvidos["limite_sla_horas"]
)

resolvidos

,id_chamado,categoria,prioridade,status,tempo_resolucao_horas,satisfacao,reaberto,limite_sla_horas,dentro_sla
0,CH001,Acesso,Alta,Resolvido,3.5,5.0,Não,8,True
1,CH002,Sistema lento,Média,Resolvido,12.0,4.0,Não,24,True
2,CH003,Impressora,Baixa,Resolvido,30.0,3.0,Sim,48,True
3,CH004,Erro de sistema,Crítica,Resolvido,2.0,5.0,Não,4,True
5,CH006,Rede,Alta,Resolvido,9.5,2.0,Sim,8,False
6,CH007,Hardware,Média,Resolvido,20.0,4.0,Não,24,True
7,CH008,Sistema lento,Baixa,Resolvido,45.0,4.0,Não,48,True
8,CH009,Erro de sistema,Alta,Resolvido,7.0,5.0,Não,8,True
9,CH010,Rede,Crítica,Resolvido,6.0,2.0,Sim,4,False
11,CH012,Acesso,Baixa,Resolvido,10.0,5.0,Não,48,True


In [29]:
contagem = resolvidos["dentro_sla"].value_counts()

contagem

,count
dentro_sla,
True,11
False,5


In [30]:
percentual = (
    resolvidos["dentro_sla"]
    .value_counts(normalize=True)
    .mul(100)
    .round(1)
)

percentual

,proportion
dentro_sla,
True,68.8
False,31.2


In [31]:
media = (
    resolvidos
    .groupby("categoria")["tempo_resolucao_horas"]
    .mean()
    .sort_values(ascending=False)
    .round(2)
)

media

,tempo_resolucao_horas
categoria,
Sistema lento,47.75
Impressora,30.00
Acesso,21.83
Hardware,18.00
Rede,11.17
Erro de sistema,4.00


A análise mostrou que 11 dos 16 chamados resolvidos ficaram dentro do SLA, correspondendo a 68,8%, enquanto 5 chamados ficaram fora, representando 31,2%. A categoria Sistema lento apresentou o maior tempo médio de resolução, com 47,75 horas, influenciada pelo chamado de 120 horas. Recomenda-se investigar os chamados de Sistema lento e os casos que ultrapassaram o SLA para identificar as causas dos atrasos. Os chamados ainda não resolvidos não foram considerados no cálculo do tempo final porque ainda não possuem um tempo de resolução concluído.